src/charm.py

```diff
@@ -133,33 +134,8 @@ def _on_grafana_changed(self, event):
 
         event.relation.data[self.unit]["sources"] = json.dumps(source_data)
 
-    def _on_alertmanager_changed(self, event):
-        """Set an alertmanager configuration.
-
-        In response to any changes in relations with Alertmanager,
-        the list of currently available Alertmanagers is updated,
-        a new Prometheus configuration set and Prometheus is
-        restarted.
-        """
-        if not self.unit.is_leader():
-            return
-
-        addrs = json.loads(event.relation.data[event.app].get("addrs", "[]"))
-
-        self._stored.alertmanagers = addrs
-
-        self._configure()
-
-    def _on_alertmanager_broken(self, event):
-        """Remove all alertmanager configuration.
-
-        When an Alertmanager departs it is removed from the list
-        of currently available Alertmanagers, the Prometheus configuration
-        is updated and Prometheus is restarted.
-        """
-        if not self.unit.is_leader():
-            return
-        self._stored.alertmanagers.clear()
+    def _on_alertmanager_cluster_changed(self, event):
+        self._stored.alertmanagers = self.alertmanager_lib.get_cluster_info()
```

Comment:
Now that you have `get_cluster_info()` it is no longer necessary to store this in two places that is in the Prometheus charm and the `AlertmanagerConsumer`. It should be possible to just invoke `get_cluster_info()` from within `_alerting_config()` method of the Prometheus charm.